### ISSUE TITLE
not able to parse pyproject.toml if contain property" package-mode=false"

### ISSUE THREAD

**Issue by siongwai** [dependabot/dependabot-core#9209]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

pip

### Manifest location and content before the Dependabot update

I am having a pyproject.toml in root level

content will be:
```
[tool.poetry]
name = "abc"
version = "0.1.0"
description = "Example"
authors = ["Your Name <you@example.com>"]
# packages = []
package-mode = false

[tool.poetry.dependencies]
python = "^3.10"
robotframework = "7.0"
robotframework-browser = "18.1.0"
robotframework-jsonlibrary = "0.5"
robotframework-pabot = "2.16.0"
robotframework-pythonlibcore = "4.3.0"
robotframework-requests = "0.9.5"
timecode = "1.3.0"

[tool.poetry.group.dev.dependencies]
robotframework-robocop = "5.0.3"
robotframework-tidy = "4.10.0"
ruff = "0.3.0"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

### dependabot.yml content

```
# Basic dependabot.yml file with
# minimum configuration package managers

version: 2
updates:
  # Configuration for pip
  - package-ecosystem: "pip"
    directory: "/"
    schedule:
      interval: "daily"
```

### What you expected to see, versus what you actually saw

Expected to see:
Able to parse pyproject.toml and create PR for monitored packages.

What I see:
dependabot return error, part of the error quote below

> updater | 2024/03/05 06:40:50 INFO <job_796061961> [Transport] Sending envelope with items [event] ebb1076325524c3db7edd5847ce2700b to Sentry
  proxy | 2024/03/05 06:40:50 [066] POST https://sentry.io:443/api/1451818/envelope/
  proxy | 2024/03/05 06:40:50 [066] 200 https://sentry.io:443/api/1451818/envelope/
updater | 2024/03/05 06:40:51 ERROR <job_796061961> Error processing robotframework-requests (Dependabot::SharedHelpers::HelperSubprocessFailed)
updater | 2024/03/05 06:40:51 ERROR <job_796061961> The Poetry configuration is invalid:
updater |   - Additional properties are not allowed ('package-mode' was unexpected)
updater | 2024/03/05 06:40:51 ERROR <job_796061961> /home/dependabot/common/lib/dependabot/shared_helpers.rb:429:in `run_shell_command'
updater | 2024/03/05 06:40:51 ERROR <job_796061961> /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11193/lib/types/private/methods/call_validation.rb:169:in `bind_call'

**Comment by mikicz** [comment]
I believe an upgrade to poetry 1.8.x should fix this

**Comment by siongwai** [comment]
Waiting for #9189 to be merged

**Comment by OldSneerJaw** [comment]
Dependabot has now automatically closed the original pull request and opened a new one: #9291. Hopefully this one gets some attention.

**Comment by abdulapopoola** [comment]
This should be fixed with #9291
